Description: Fix --mtu on recent kernels
 This patch allows EMSGSIZE replies with SO_EE_ORIGIN_LOCAL so that
 traceroute with the --mtu option can work on recent kernels
Author: Graham Inggs <graham@nerve.org.za>
Forwarded: https://sourceforge.net/p/traceroute/bugs/8/
Last-Update: 2014-08-09
--- a/traceroute/traceroute.c
+++ b/traceroute/traceroute.c
@@ -1345,7 +1345,8 @@
 
 		    ee = (struct sock_extended_err *) ptr;
 
-		    if (ee->ee_origin != SO_EE_ORIGIN_ICMP)
+		    if ((ee->ee_origin != SO_EE_ORIGIN_ICMP) &&
+			(ee->ee_origin != SO_EE_ORIGIN_LOCAL))
 			    ee = NULL;
 		    /*  dgram icmp sockets might return extra things...  */
 		    else if (ee->ee_type == ICMP_SOURCE_QUENCH ||
